platform-core-module/src/main/java/com/softicar/platform/core/module/program/QueuedProgramExecutionDaemon.java

```diff
@@ -102,6 +116,29 @@ private void terminate(Collection<IRunnableThread<ProgramExecutionRunnable>> run
 		}
 	}
 
+	private void checkMaximumRuntime(AGProgram program, AGProgramExecution currentExecution) {
+
+		var scheduledExecution = AGScheduledProgramExecution.getByAGUuid(program.getProgramUuid());
+		if (scheduledExecution != null) {
+			var maximumRuntimeInSeconds = scheduledExecution.getMaximumRuntimeInSeconds();
+			if (maximumRuntimeInSeconds.isPresent()) {
+				boolean maximumRuntimeExceeded = DayTime.now().isAfter(currentExecution.getStartedAt().plusSeconds(maximumRuntimeInSeconds.get()));
+				if (maximumRuntimeExceeded && !currentExecution.isExceededMaximumRuntime()) {
```

Comment:
Had to read this line several times. It is very confusing.
```suggestion
				if (maximumRuntimeExceeded && !currentExecution.isExceededRuntimeReported()) {
```

Reply:
hm I dont think so, maybe a second opinion would be great :)

Reply:
another suggestion would be `isExceededRuntimeHandled`